### PR TITLE
Fix Windows connection reset detection

### PR DESCRIPTION
**Pull request by nicktindall**
Fix a test failing in Windows. increase overall hackiness.

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/network/internal/SocketExceptionUtil.java [hunk 1]
```diff
@@ -20,9 +20,20 @@ private SocketExceptionUtil() {
      * @return true if the exception is one that signifies the connection was reset
      */
     public static boolean isAConnectionResetException(IOException e) {
-        String message = e.getMessage();
-        return message != null &&
-                (message.equals("Connection reset by peer")
-                        || e instanceof SocketException && message.equals("Connection reset"));
+        return isALinuxJava12OrLessConnectionResetException(e)
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/network/internal/SocketExceptionUtil.java:23]: my word!

**Review by JerryShea — APPROVED** [review]
very good indeed!

File: src/main/java/net/openhft/chronicle/network/internal/SocketExceptionUtil.java (hunk 1 above)
Comment (JerryShea) [reply src/main/java/net/openhft/chronicle/network/internal/SocketExceptionUtil.java:23]: Write Once Run Anywhere? ;)

Comment (nicktindall) [reply src/main/java/net/openhft/chronicle/network/internal/SocketExceptionUtil.java:23]: Annoying. Connection reset seems like a strong enough concept to warrant its own IOException subclass.

**Pull request merged**